app/controllers/users_controller.rb

```diff
@@ -0,0 +1,50 @@
+class UsersController < ApplicationController
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'users/new'
+    end
+  end
+
+  post '/signup' do
+    @user = User.new(params)
+    if @user.save
+      log_in_user
+      redirect '/tweets'
+    else
+      redirect '/signup'
+    end
+  end
+
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    @tweets = Tweet.all { |tweet| tweet.user_id == @user.id }
```

Comment:
Here again, you should be able to use `@user.tweets` instead of checking the foreign key. Also, it is preferable here to use ActiveRecord's `where` method to filter results on the database side, rather than processing the results in Ruby.